### PR TITLE
docs: added how to connect/authenticate, how to call methods, how to close connection

### PR DESCRIPTION
**Pull request by KAVAN-DESAI**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -25,23 +25,75 @@
...
-```dart
-import 'package:cln_grpc/cln_grpc.dart';
+To connect to a grpc server we need port of server, host and ChannelOptions
 
-Future<void> main(List<String> args) async {
-  var client = GRPCClient(rootPath: args[0], certClientPath: "/");
+Here in this libraray the only required parameter is path to certificates for authentification, The library also provides some optional parameters such as 
+
+- host("localhost" by default)
+- port(8001 by default)
```
Comment (vincenzopalazzo) [inline README.md:43]:
```suggestion
- port(8001 by default): where the grpc server is running, specified as `grpc-port` in the core lightning option
```

File: README.md [hunk 2]
```diff
@@ -25,23 +25,75 @@
...
+- authority("localhost" by default)
```
Comment (vincenzopalazzo) [inline README.md:44]:
```suggestion
- authority("localhost" by default)
```

This I don't know what it is :) we need to check in the dart docs hahaa

File: README.md [hunk 3]
```diff
@@ -25,23 +25,75 @@
...
+- channel options
```
Comment (vincenzopalazzo) [inline README.md:45]: same observation of before

File: README.md [hunk 4]
```diff
@@ -25,23 +25,75 @@
...
+
+To inititate the client and make a connection with the server we declare client like this:
+```dart
+var client = GRPCClient(rootPath: "<path_to_certificates>");
+```
+### How to call a method:
+There are several ways to call a method 
+1. call getinfo with a parse function
+```dart
   var response = await client.getinfo();
```
Comment (vincenzopalazzo) [inline README.md:55]:
```suggestion
  var response = await client.getinfo(onDecode: (Map<String, dynamc> json) => json);
```

File: README.md [hunk 5]
```diff
@@ -25,23 +25,75 @@
...
+  ListTransactionProxy(this.proxy);
+
+  factory ListTransactionProxy.build() =>
+      ListTransactionProxy(ListtransactionsRequest());
+
+  @override
+  Map<String, dynamic> toJSON() => proxy.writeToJsonMap();
+
+  @override
+  T as<T>() => proxy as T;
```
Comment (vincenzopalazzo) [inline README.md:74]: we need to check if we can remove this requirement

File: README.md [hunk 6]
```diff
@@ -25,23 +25,75 @@
...
+```dart
+  var forwardsList=await client.stub.listForwards(ListforwardsRequest());
+  print("forwards of node");
+  print(forwardsList.forwards);
+```
+### Close client connection:
+Shutting down the grpc client connection is really easy calling the close method like this:
+```dart
+  client.close();
+```
```
Comment (vincenzopalazzo) [inline README.md:96]: we can put this example in some different dart file in the example dir? in this way we ca test it? also in some example when you say "with the parse funtion" you are missing the parse function as parameter

**Review by vincenzopalazzo — CHANGES_REQUESTED** [review]

File: README.md [hunk 7]
```diff
@@ -25,23 +25,75 @@
...
 
-- clightning_rpc: The RPC wrapper around the core lightning API.
-- cln_plugin: A library to write extensible plugin in dart for core lightning.
 
 ## How to Use
+### How to connect to core lightning using cln_grpc
 
-```dart
-import 'package:cln_grpc/cln_grpc.dart';
+To connect to a grpc server we need port of server, host and ChannelOptions
```
[the remark this answers is not in the document]
Comment (KAVAN-DESAI) [reply README.md:38]: oops, its channel credentials ```ChannelCredentials opts;```

File: README.md (hunk 6 above)
Comment (KAVAN-DESAI) [reply README.md:96]: Ya! I was going to raise this issue to create all different types of examples

File: README.md (hunk 2 above)
Comment (vincenzopalazzo) [reply README.md:44]: yeah! good documented eh?

Lets skip this and left as is

File: README.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (vincenzopalazzo) [reply README.md:38]: maybe we can say more general, you can specify the server credential with the `ChannelCredentials` constructor option, or something like that?

File: README.md (hunk 6 above)
Comment (KAVAN-DESAI) [reply README.md:96]: So, should I skip (2,3) in documentation? and add some example files?

Comment (vincenzopalazzo) [reply README.md:97]: no keep also in the docs, but add the link to the example :) if you already remove it, it is fine too!

Comment (vincenzopalazzo) [reply README.md:97]: yes! you add the link to the file :)

**Review by vincenzopalazzo — APPROVED** [review]
Good for me.

We land this PR on master and we keep track of this issue https://github.com/dart-lightning/lndart.cln_grpc/issues/20

**Pull request merged**